src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs

```diff
@@ -239,50 +268,90 @@ internal class SafeAreaDetails
 			private readonly Thickness _originalMargin;
 			private Thickness _appliedPadding = new Thickness(0);
 			private Thickness _appliedMargin = new Thickness(0);
+			private readonly CompositeDisposable _subscriptions = new();
 
 			internal SafeAreaDetails(FrameworkElement owner)
 			{
 				_owner = new WeakReference(owner);
 
 				_originalMargin = owner.Margin;
 				_originalPadding = PaddingHelper.GetPadding(owner);
-
-				_visibleBoundsChanged = (s2, e2) => UpdateInsets();
 #if __IOS__
 				// For iOS, it's required to react on SizeChanged to prevent weird alignment
 				// problems with Text using the LayoutManager (NSTextContainer).
 				// https://github.com/unoplatform/uno/issues/2836
 
-				owner.SizeChanged += (s, e) => UpdateInsets();
+				owner.SizeChanged += OnInsetUpdateRequired;
+				Disposable.Create(() => owner.SizeChanged -= OnInsetUpdateRequired)
+					.DisposeWith(_subscriptions);
 #endif
-				owner.LayoutUpdated += (s, e) => UpdateInsets();
-				owner.Loaded += (s, e) =>
+				owner.LayoutUpdated += OnInsetUpdateRequired;
+				Disposable.Create(() => owner.LayoutUpdated -= OnInsetUpdateRequired)
+					.DisposeWith(_subscriptions);
+
+				if (!owner.IsLoaded)
 				{
-					UpdateInsets();
-					ApplicationView.GetForCurrentView().VisibleBoundsChanged += _visibleBoundsChanged;
+					owner.Loaded += OnOwnerLoaded;
+					Disposable.Create(() => owner.Loaded -= OnOwnerLoaded)
+					.DisposeWith(_subscriptions);
+				}
+				else
+				{
+					RegisterEvents();
+				}
 
-					if (InputPane.GetForCurrentView() is { } inputPane)
-					{
-						inputPane.Showing += OnInputPaneChanged;
-						inputPane.Hiding += OnInputPaneChanged;
-					}
-				};
-				owner.Unloaded += (s, e) =>
+				owner.Unloaded += OnOwnerUnloaded;
+				Disposable.Create(() => owner.Unloaded -= OnOwnerUnloaded)
+					.DisposeWith(_subscriptions);
+			}
+
+			private void OnOwnerUnloaded(object sender, RoutedEventArgs e)
+			{
+				Unregister();
+			}
+
+			private void OnOwnerLoaded(object sender, RoutedEventArgs e)
+			{
+				UpdateInsets();
+				RegisterEvents();
+			}
+
+			private void OnInsetUpdateRequired(object? sender, object? args)
+			{
+				UpdateInsets();
+			}
+
+			private void RegisterEvents()
+			{
+				ApplicationView.GetForCurrentView().VisibleBoundsChanged += OnInsetUpdateRequired;
+				Disposable.Create(() => ApplicationView.GetForCurrentView().VisibleBoundsChanged -= OnInsetUpdateRequired)
+					.DisposeWith(_subscriptions);
+
+
+				if (InputPane.GetForCurrentView() is { } inputPane)
 				{
-					if (InputPane.GetForCurrentView() is { } inputPane)
-					{
-						inputPane.Showing -= OnInputPaneChanged;
-						inputPane.Hiding -= OnInputPaneChanged;
-					}
-				};
+					inputPane.Showing += OnInputPaneChanged;
+					inputPane.Hiding += OnInputPaneChanged;
+
+					Disposable.Create(() =>
+						{
+							inputPane.Showing -= OnInputPaneChanged;
+							inputPane.Hiding -= OnInputPaneChanged;
+						})
+					.DisposeWith(_subscriptions);
```

Comment:
thats invalid indentation; pick one:
```cs
Disposable.Create(() =>
{
	inputPane.Showing -= OnInputPaneChanged;
	inputPane.Hiding -= OnInputPaneChanged;
}).DisposeWith(_subscriptions);
// OR
Disposable
	.Create(() =>
	{
		inputPane.Showing -= OnInputPaneChanged;
		inputPane.Hiding -= OnInputPaneChanged;
	})
	.DisposeWith(_subscriptions);
```
the first can even be augmented with: _subscriptions.Add(() => ...)